File: research/agent-skills-guard-main/src/types/featured-marketplace.ts
export interface LocalizedText {
  en: string;
  zh: string;
}

export interface FeaturedMarketplacesConfig {
  version: string;
  last_updated: string;
  marketplace: FeaturedMarketplaceCategory[];
}

export interface FeaturedMarketplaceCategory {
  id: string;
  name: LocalizedText;
  description: LocalizedText;
  marketplaces: FeaturedMarketplace[];
}

export interface FeaturedMarketplaceOwner {
  name?: string;
  email?: string;
}

export interface FeaturedMarketplace {
  marketplace_name: string;
  marketplace_repo: string;
  repository_url?: string;
  marketplace_add_command?: string;
  description: LocalizedText;
  owner?: FeaturedMarketplaceOwner;
  tags: string[];
  featured: boolean;
  plugins: FeaturedMarketplacePlugin[];
}

export interface FeaturedMarketplacePlugin {
  name: string;
  install_command?: string;
  description: LocalizedText;
  version?: string;
  author?: FeaturedMarketplaceOwner;
  source?: string;
  tags: string[];
  featured: boolean;
}

File: research/agent-skills-guard-main/src-tauri/src/models/featured_marketplace.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LocalizedText {
    pub en: String,
    pub zh: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FeaturedMarketplacesConfig {
    pub version: String,
    pub last_updated: String,
    #[serde(rename = "marketplace", alias = "categories")]
    pub marketplace: Vec<FeaturedMarketplaceCategory>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FeaturedMarketplaceCategory {
    pub id: String,
    pub name: LocalizedText,
    pub description: LocalizedText,
    pub marketplaces: Vec<FeaturedMarketplace>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FeaturedMarketplaceOwner {
    pub name: Option<String>,
    pub email: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FeaturedMarketplace {
    pub marketplace_name: String,
    pub marketplace_repo: String,
    pub repository_url: Option<String>,
    pub marketplace_add_command: Option<String>,
    pub description: LocalizedText,
    pub owner: Option<FeaturedMarketplaceOwner>,
    pub tags: Vec<String>,
    pub featured: bool,
    pub plugins: Vec<FeaturedMarketplacePlugin>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FeaturedMarketplacePlugin {
    pub name: String,
    pub install_command: Option<String>,
    pub description: LocalizedText,
    pub version: Option<String>,
    pub author: Option<FeaturedMarketplaceOwner>,
    pub source: Option<String>,
    pub tags: Vec<String>,
    pub featured: bool,
}
